### ISSUE TITLE
output the failed CR from scorecard basic test

### ISSUE THREAD

**Issue by jmrodri** [operator-framework/operator-sdk#5212]
#### Describe the problem you need a feature to resolve.

The basic scorecard test fails simply stating `error spec does not exist` but it does not tell you which CR is affected.

```
$ operator-sdk scorecard bundle --selector test=basic-check-spec-test --verbose
DEBU[0000] Debug logging is set                         
--------------------------------------------------------------------------------
Image:      quay.io/operator-framework/scorecard-test:v1.10.1
Entrypoint: [scorecard-test basic-check-spec]
Labels:
    "suite":"basic"
    "test":"basic-check-spec-test"
Results:
    Name: basic-check-spec
    State: fail
    Errors:
        error spec does not exist
```

#### Describe the solution you'd like.

`scorecard` should output the failed CR as part of the error. It has this information when the error is created.

```go
func checkSpec(crSet []unstructured.Unstructured,
	res scapiv1alpha3.TestResult) scapiv1alpha3.TestResult {
	for _, cr := range crSet {
		if cr.Object["spec"] == nil {
			res.Errors = append(res.Errors, "error spec does not exist")
			res.State = scapiv1alpha3.FailState
			return res
		}
	}
	return res
}
```

maybe something like

```
$ operator-sdk scorecard bundle --selector test=basic-check-spec-test --verbose
DEBU[0000] Debug logging is set                         
--------------------------------------------------------------------------------
Image:      quay.io/operator-framework/scorecard-test:v1.10.1
Entrypoint: [scorecard-test basic-check-spec]
Labels:
    "suite":"basic"
    "test":"basic-check-spec-test"
Results:
    Name: basic-check-spec
    State: fail
    Errors:
        error spec for "cache.memcached.yaml" does not exist
```

/language go

**Comment by jmrodri** [comment]
@thomasmckay here's the issue to cover the item we talked about in slack.

**Comment by jmrodri** [comment]
Fixed by #5255.
